files/default/audit_report.rb

```diff
@@ -98,6 +116,22 @@ def gather_nodeinfo
         }
       end
 
+      # this is a helper methods to extract the profiles we scan and hand this
+      # over to the reporter in addition to the `json-min` report. `json-min`
+      # reports do not include information about the source of the profiles
+      # TODO: should be available in inspec `json-min` reports out-of-the-box
+      # TODO: raise warning when not a compliance-known profile
+      def cc_profile_index(profiles)
+        cc_profiles = tests_for_runner(profiles).map { |profile| profile[:compliance] if profile[:compliance] }.uniq.compact
```

Comment:
I think this could be `tests_for_runner(profiles).select { |profile| profile[:compliance] }.uniq.compact`:

```
irb(main):001:0> ls = [ { foo: nil }, { bar: 3 }, { foo: 4 }]
=> [{:foo=>nil}, {:bar=>3}, {:foo=>4}]
irb(main):002:0> ls.select {|x| x[:foo] }
=> [{:foo=>4}]
```